client/src/components/admin/CompaniesList.vue

```diff
@@ -0,0 +1,346 @@
+                <input type="text" class="form-control" id="address"
+                       aria-describedby="address" placeholder="Straße"
+                       v-model="companies[currentEditCompanyIndex].address.street" />
+              </div>
+
+              <div class="col-6">
+                <input type="text" class="form-control"
+                       aria-describedby="number" placeholder="Nummer"
+                       v-model="companies[currentEditCompanyIndex].address.number" />
+              </div>
+            </div>
+
+            <div class="mb-2 row g-2 align-items-center">
+              <div class="col-6">
+                <input type="text" class="form-control"
+                       aria-describedby="zip" placeholder="PLZ"
+                       v-model="companies[currentEditCompanyIndex].address.zip" />
+              </div>
+
+              <div class="col-6">
+                <input type="text" class="form-control"
+                       aria-describedby="city" placeholder="Ort"
+                       v-model="companies[currentEditCompanyIndex].address.city" />
+              </div>
+
+              <div class="col-12">
+                <input type="text" class="form-control"
+                       aria-describedby="city" placeholder="Land"
+                       v-model="companies[currentEditCompanyIndex].address.country" />
+              </div>
+
+              <div class="col-6">
+                <input type="text" class="form-control"
+                       aria-describedby="latitude" placeholder="Latitude"
+                       v-model="companies[currentEditCompanyIndex].address.latitude" />
+              </div>
+
+              <div class="col-6">
+                <input type="text" class="form-control"
+                       aria-describedby="longitude" placeholder="Longitude"
+                       v-model="companies[currentEditCompanyIndex].address.longitude" />
+              </div>
+            </div>
+          </div>
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-secondary"
+                  data-bs-dismiss="modal">Schließen</button>
+          <button type="button"
+                  class="btn btn-success"
+                  @click="updateCompany(currentEditCompanyIndex)"
+          >Speichern</button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import { getCompaniesList } from '@/utils/gateways';
+import Company from '@/models/Company';
+import store from '@/store';
+
+export default defineComponent({
+  name: 'CompaniesList',
+  data() {
+    return {
+      currentEditCompanyIndex: 0,
+      companies: [] as Company[],
+      currentSorting: '',
+    };
+  },
+  mounted() {
+    this.updateList();
+  },
+  methods: {
+    updateList() {
+      // API call for GET list with params
+      getCompaniesList()
+        .then((list) => {
```

Comment:
eventuel noch auf async/await anpassen

Reply:
was hat das in dem fall für einen vorteil?

Reply:
Es vermeidet Callback-Hell :D